cmd/main.go

```diff
@@ -84,6 +91,71 @@ func main() {
 		l.Fatalf("Enclave terminated: %v", err)
 	}
 
-	// Block on this read forever.
-	<-make(chan struct{})
+	// Nitriding supports two ways of starting the enclave application:
+	//
+	// 1) Nitriding spawns the enclave application itself, and waits for it
+	//    to terminate.
+	//
+	// 2) The enclave application is started by a shell script (which also
+	//    starts nitriding).  In this case, we simply block forever.
+	if appCmd != "" {
+		f := func(s string) {
+			l.Printf("Application says: %s", s)
+		}
+		runAppCommand(appCmd, f, f)
+	} else {
+		// Block forever.
+		<-make(chan struct{})
+	}
+	l.Println("Exiting nitriding.")
+}
+
+// runAppCommand (i) runs the given command, (ii) waits until the command
+// finished execution, and (iii) in the meanwhile prints the command's stdout
+// and stderr.
+func runAppCommand(appCmd string, stdoutFunc, stderrFunc func(string)) {
+	l.Printf("Invoking the enclave application.")
+	args := strings.Split(appCmd, " ")
+	cmd := exec.Command(args[0], args[1:]...)
+
+	// Print the enclave application's stderr.
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		l.Fatalf("Failed to obtain stderr pipe for enclave application: %v", err)
+	}
+	go printOutput(stderr, stderrFunc, "stderr")
+
+	// Print the enclave application's stdout.
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		l.Fatalf("Failed to obtain stdout pipe for enclave application: %v", err)
+	}
+	go printOutput(stdout, stdoutFunc, "stdout")
+
+	if err := cmd.Start(); err != nil {
+		l.Fatalf("Failed to start enclave application: %v", err)
+	}
+
+	if err := cmd.Wait(); err != nil {
+		l.Fatalf("Enclave application exited with non-0 exit code: %v", err)
+	}
+	l.Println("Enclave application exited.")
+}
+
+// printOutput continuously reads from the given Reader until an EOF occurs.
+// Each newly read line is passed to the given function f.
+func printOutput(readCloser io.ReadCloser, f func(string), output string) {
+	r := bufio.NewReader(readCloser)
+	for {
+		b, err := r.ReadBytes(0x0a) // Read until we see a newline.
+		if errors.Is(err, io.EOF) {
+			l.Printf("Encountered EOF in %s.  Returning.", output)
+			return
```

Comment:
I think this drops a final, unterminated line. Should still forward `f(b)` in this case?
> ReadBytes reads until the first occurrence of delim in the input, returning a slice containing the data up to and including the delimiter. If ReadBytes encounters an error before finding a delimiter, it returns the data read before the error and the error itself (often io.EOF). ReadBytes returns err != nil if and only if the returned data does not end in delim.

Reply:
The same docs also suggest Scanner for line breaking.
```go
scanner := bufio.NewScanner(readCloser)
for scanner.Scan() {
    f(scanner.Text()) // f() needs to use Println since the scanner absorbs the delimeter.
}
if err := scanner.Err(); err != nil {
    l.Printf("Error reading from enclave application:", err)
}
```